src/backend/distributed/utils/multi_transaction.c

```diff
@@ -347,3 +348,62 @@ ConnectionList(HTAB *connectionHash)
 
 	return connectionList;
 }
+
+
+/*
+ * OpenConnectionsToShardPlacements opens connections to all placements of the
+ * shard with the given shardId and populates the shardConnectionHash table
+ * accordingly.
+ */
+void
+OpenConnectionsToShardPlacements(uint64 shardId, HTAB *shardConnectionHash,
```

Comment:
What do you think of moving this into a transaction/shard_transaction.c file?

In the future, we'll have a worker_transaction.c related to doing transactions across all workers (e.g. syncing metadata), which uses the 2PC functions from multi_transaction.c. It would be nice if we separated logic for doing transactions across shard placements into a separate file.